### PR TITLE
Adds postgres 14 docker template for update_docker

### PR DESCRIPTION
**Pull request by gurkanindibay**
Recently we added pg 14 support however we haven't  added the template to be used by update_docker
This template is being added by this PR

Error is as below

### CODE REVIEW

File: packaging_automation/update_docker.py [hunk 1]
```diff
@@ -63,6 +66,17 @@ def update_docker_file_for_postgres12(project_version: str, template_path: str,
     write_to_file(content, dest_file_name)
 
 
+def update_docker_file_for_postgres14(project_version: str, template_path: str, exec_path: str):
+    content = process_template_file(project_version, template_path,
+                                    docker_templates[SupportedDockerImages.postgres14])
+    dest_file_name = f"{exec_path}/{docker_outputs[SupportedDockerImages.postgres14]}"
+    dir_name=os.path.dirname(dest_file_name)
+    if not (os.path.exists(dir_name)):
```
Comment (onurctirtir) [inline packaging_automation/update_docker.py:74]: very minor:

```suggestion
    if not os.path.exists(dir_name):
```

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+ENV CITUS_VERSION ${VERSION}.citus-1
+
+# install Citus
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+       ca-certificates \
+       curl \
+    && curl -s https://install.citusdata.com/community/deb.sh | bash \
+    && apt-get install -y postgresql-$PG_MAJOR-citus-10.2=$CITUS_VERSION \
```
Comment (onurctirtir) [inline packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:23]:
```suggestion
    && apt-get install -y postgresql-$PG_MAJOR-citus-{{project_minor_version}}.=$CITUS_VERSION \
```

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+
+# This file is auto generated from it's template,
+# see citusdata/tools/packaging_automation/templates/docker/latest/latest.tmpl.dockerfile.
```
Comment (onurctirtir) [inline packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:3]:
```suggestion
# see citusdata/tools/packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile.
```

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+# install Citus
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+       ca-certificates \
+       curl \
+    && curl -s https://install.citusdata.com/community/deb.sh | bash \
+    && apt-get install -y postgresql-$PG_MAJOR-citus-10.2=$CITUS_VERSION \
+                          postgresql-$PG_MAJOR-hll=2.16.citus-1 \
+                          postgresql-$PG_MAJOR-topn=2.4.0 \
```
Comment (onurctirtir) [inline packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:25]: We are using newest hll & topn versions for pg14, this is nice.

But I realized that we use different versions of those two in pg12 image, can you look into that separately ?

File: packaging_automation/update_docker.py [hunk 5]
```diff
@@ -63,6 +66,17 @@ def update_docker_file_for_postgres12(project_version: str, template_path: str,
     write_to_file(content, dest_file_name)
 
 
+def update_docker_file_for_postgres14(project_version: str, template_path: str, exec_path: str):
+    content = process_template_file(project_version, template_path,
+                                    docker_templates[SupportedDockerImages.postgres14])
+    dest_file_name = f"{exec_path}/{docker_outputs[SupportedDockerImages.postgres14]}"
+    dir_name=os.path.dirname(dest_file_name)
```
Comment (onurctirtir) [inline packaging_automation/update_docker.py:73]: very minor:

```suggestion
    dir_name = os.path.dirname(dest_file_name)
```

**Review by onurctirtir — APPROVED** [review]
approving with some minor comments

File: packaging_automation/update_docker.py (hunk 1 above)
Comment (gurkanindibay) [reply packaging_automation/update_docker.py:74]: Fixed

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile (hunk 2 above)
Comment (gurkanindibay) [reply packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:23]: Fixed

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile (hunk 3 above)
Comment (gurkanindibay) [reply packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:3]: Fixed

File: packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile (hunk 4 above)
Comment (gurkanindibay) [reply packaging_automation/templates/docker/postgres-14/postgres-14.tmpl.dockerfile:25]: Fixed

File: packaging_automation/update_docker.py (hunk 5 above)
Comment (gurkanindibay) [reply packaging_automation/update_docker.py:73]: Fixed

**Pull request merged**